Bottom: 570394b2881f527f944f23db393e32799976b5d5
Top:    65ddf37285622facc3bf0ede05f69bfee2d85567
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-21 15:00:20 +0000

Refresh of try-debugging-draw-speed

---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 38c1ca2..b6cf8bb 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -570,6 +570,28 @@ ghid_invalidate_all ()
   gdk_window_invalidate_rect (gport->drawing_area->window, NULL, 1);
 }
 
+void clear_stencil (void)
+{
+#if 1
+  glStencilMask (~0);
+  glClear (GL_STENCIL_BUFFER_BIT);
+#else
+  glPushMatrix ();
+  glLoadIdentity ();
+  glStencilFunc (GL_ALWAYS, 0, ~0);   // Always pass stencil test, ref=0
+  glColorMask (0, 0, 0, 0);           // Disable writting in color buffer
+  glBegin (GL_QUADS);
+  glVertex2i (0, 0);
+  glVertex2i (0, gport->height);
+  glVertex2i (gport->width, gport->height);
+  glVertex2i (gport->width, 0);
+  glEnd ();
+  glStencilFunc (GL_GREATER, 1, 1);   // Back to the normal stencil test
+  glColorMask (1, 1, 1, 1);           // Enable writting in color buffer
+  glPopMatrix ();
+#endif
+}
+
 
 int
 ghid_set_layer (const char *name, int group, int empty)
@@ -581,7 +603,8 @@ ghid_set_layer (const char *name, int group, int empty)
   /* Reset stencil buffer so we can paint anywhere */
   hidgl_flush_triangles (&buffer);
 //  glClear (GL_STENCIL_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
-  glClear (GL_STENCIL_BUFFER_BIT);
+//  glClear (GL_STENCIL_BUFFER_BIT);
+  clear_stencil ();
 
   if (idx >= 0 && idx < max_layer + 2) {
     gport->trans_lines = TRUE;
